src/cls/rgw_gc/cls_rgw_gc_client.h

```diff
@@ -0,0 +1,17 @@
+#ifndef CEPH_CLS_RGW_GC_CLIENT_H
+#define CEPH_CLS_RGW_GC_CLIENT_H
+
+#include "include/rados/librados.hpp"
+#include "cls/rgw_gc/cls_rgw_gc_types.h"
+#include "cls/queue/cls_queue_ops.h"
+#include "common/ceph_time.h"
+
+void cls_rgw_gc_queue_init(librados::ObjectWriteOperation& op, const string& queue_name, uint64_t size, uint64_t num_urgent_data_entries);
+int cls_rgw_gc_queue_get_capacity(librados::IoCtx& io_ctx, const string& oid, uint64_t& size);
+void cls_rgw_gc_queue_enqueue(librados::ObjectWriteOperation& op, uint32_t expiration_secs, const cls_rgw_gc_obj_info& info);
+int cls_rgw_gc_queue_list_entries(librados::IoCtx& io_ctx, const string& oid, const string& marker, uint32_t max, bool expired_only,
+                    list<cls_rgw_gc_obj_info>& entries, bool *truncated, string& next_marker);
+void cls_rgw_gc_queue_remove_entries(librados::ObjectWriteOperation& op, uint32_t num_entries);
+void cls_rgw_gc_queue_defer_entry(librados::ObjectWriteOperation& op, uint32_t expiration_secs, const cls_rgw_gc_obj_info& info);
```

Comment:
this new `cls_rgw_gc_queue_defer_entry()` method takes `cls_rgw_gc_obj_info` (which is a tag and a chain), where the existing `cls_rgw_gc_defer_entry()` takes only the tag. it doesn't look like this new op actually uses the chain part, so can we change this to match?

Reply:
As part of gc_queue_defer_entry, we do the following:
1. Check whether the tag exists in urgent data, update it if it exists, else add tag and expiration time to urgent data.
2. Enqueue the entry -> This step requires info (both chain and tag)

Reply:
ok thanks, i missed that part!